pkg/controller/operators/olm/operator.go

```diff
@@ -61,6 +61,10 @@ var (
 	ErrAPIServiceOwnerConflict = errors.New("unable to adopt APIService")
 )
 
+// this unexported operator plugin slice provides an entrypoint for
+// downstream to inject its own plugins to augment the controller behavior
+var operatorPlugInFactoryFuncs []plugins.OperatorPlugInFactoryFunc
```

Comment:
Using factory functions instead of the instantiated plug-in objects themselves. This guarantees that each instance of the operator gets their own instance of the plug-ins